From 48b949644e51867aa476ba0584be758c8aeb908d Mon Sep 17 00:00:00 2001
From: "Shoutao.Li" <Shoutao.Li@windriver.com>
Date: Tue, 14 Jun 2016 15:47:17 +0800
Subject: [PATCH] weaken dependency on ubus

If failed to connect to ubus, the rest are still executed.
And Add ubus connect debug log.

Signed-off-by: Shoutao.Li <Shoutao.Li@windriver.com>
---
 main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/main.c b/main.c
index 71463ae..8ea8684 100644
--- a/main.c
+++ b/main.c
@@ -79,7 +79,7 @@ build_state(bool runtime)
 	else
 	{
 		if (!fw3_ubus_connect())
-			error("Failed to connect to ubus");
+			warn("Failed to connect to ubus");
 
 		if (uci_load(state->uci, "firewall", &p))
 		{
--
---
 ubus.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/ubus.c b/ubus.c
index 9e034c8..bfc68c2 100644
--- a/ubus.c
+++ b/ubus.c
@@ -46,13 +46,22 @@ fw3_ubus_connect(void)
 	struct blob_buf b = { };
 
 	if (!ctx)
+	{
+		warn("ubus_connect() failed!!!");
 		goto out;
+	}
 
 	if (ubus_lookup_id(ctx, "network.interface", &id))
+	{
+		warn("ubus_lookup_id() failed!!!");
 		goto out;
+	}
 
 	if (ubus_invoke(ctx, id, "dump", NULL, dump_cb, NULL, 500))
+	{
+		warn("ubus_invoke() failed!!!");
 		goto out;
+	}
 
 	status = true;
 
-- 
2.7.4

